spec/support/test_db.rb

```diff
@@ -0,0 +1,146 @@
+
+  def self.build
+    db = self.new
+    db.cleanup_database_users
+    db.create_user_tablespace
+    db.setup_database_users
+    db.connection.logoff
+  end
+
+  def self.database_version
+     db = self.new
+     db.database_version
+  end
+
+  def connection
+    unless defined?(@connection)
+      begin
+        Timeout::timeout(5) {
+          if defined?(JRUBY_VERSION)
+            @connection = java.sql.DriverManager.get_connection(
+              'jdbc:oracle:thin:@127.0.0.1:1521/XE',
+              'system',
+              'oracle'
+            );
+          else
+            @connection = OCI8.new(
+              'system',
+              'oracle',
+              '(DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=127.0.0.1)(PORT=1521))(CONNECT_DATA=(SERVICE_NAME=XE)))'
+            )
+          end
+        }
+      rescue Timeout::Error
+        raise "Cannot establish connection with Oracle database as SYSTEM user. Seams you need to start local Oracle database"
+      end
+    end
+    @connection
+  end
+
+  def create_user_tablespace
+    return unless connection
+    execute_statement(<<-STATEMENT
+      DECLARE
+        v_exists number;
+      BEGIN
+        SELECT count(1)
+          INTO v_exists
+          FROM dba_tablespaces
+          WHERE tablespace_name = 'TBS_USERS';
+
+          IF v_exists = 0 THEN
+            EXECUTE IMMEDIATE 'CREATE TABLESPACE TBS_USERS DATAFILE ''tbs_users.dat'' SIZE 10M REUSE AUTOEXTEND ON NEXT 10M MAXSIZE 200M';
+          END IF;
+      END;
+      STATEMENT
+      )
+  end
+
+  def cleanup_database_users
+    return unless connection
+    DATABASE_USERS.each do |db|
+      execute_statement(<<-STATEMENT
+        DECLARE
+           v_count INTEGER := 0;
+           l_cnt   INTEGER;
+        BEGIN
+
+          SELECT COUNT (1)
+            INTO v_count
+            FROM dba_users
+            WHERE username = UPPER('#{db}');
+
+          IF v_count != 0 THEN
+            FOR x IN (SELECT *
+                        FROM v$session
+                        WHERE username = UPPER('#{db}'))
+            LOOP
+              EXECUTE IMMEDIATE 'ALTER SYSTEM DISCONNECT SESSION ''' || x.sid || ',' || x.serial# || ''' IMMEDIATE';
+            END LOOP;
+
+            EXECUTE IMMEDIATE ('DROP USER #{db} CASCADE');
+          END IF;
+        END;
+        STATEMENT
+      )
+    end
+  end
+
+  def setup_database_users
+    return unless connection
+    DATABASE_USERS.each do |db|
+      execute_statement(<<-STATEMENT
+        DECLARE
+           v_count INTEGER := 0;
+        BEGIN
+
+          SELECT COUNT (1)
+            INTO v_count
+            FROM dba_users
+            WHERE username = UPPER ('#{db}');
+
+          IF v_count = 0 THEN
+            EXECUTE IMMEDIATE ('CREATE USER #{db} IDENTIFIED BY #{db}');
+            EXECUTE IMMEDIATE ('GRANT unlimited tablespace, create session, create table, create sequence, create procedure, create type, create view, create synonym TO #{db}');
```

Comment:
You could use:
create user ... identified by ... default tablespace USERS quota unlimited ON USERS;